lib/Components/Search.dart

```diff
@@ -0,0 +1,24 @@
+
+import 'package:flutter/material.dart';
+import 'package:flutter_search_bar/flutter_search_bar.dart';
+
+final viewNotesScaffoldKey = GlobalKey<ScaffoldState>();
+
+// class Menu extends StatefulWidget {
```

Comment:
Do we need to leave this commented out or can we delete it?

Reply:
We can totally do without this because it is only used once. 

However, to give you some insight, the reason I would prefer it be out here is that when we declare a variable we are telling the CPU to reserve memory for it and by declaring outside the for loop, we can use that same memory as much as possible. 
However, if I was to declare it in the for loop, the CPU will have to reserve N different memory location for the same-named variable; one for each loop cycle; using up computational resources (of course flutter might be smart enough to manage that - but in system-critical software where we do not have the luxury of smart IDEs, the inefficient use of resources will be more obvious),